mongomock/helpers.py

```diff
@@ -48,7 +48,74 @@ def _index_list(key_or_list, direction=None):
                             "key_or_list must be an instance of list")
     return key_or_list
 
-  
+
+class hashdict(dict):
+    """
+    hashable dict implementation, suitable for use as a key into
+    other dicts.
+
+    >>> h1 = hashdict({"apples": 1, "bananas":2})
+    >>> h2 = hashdict({"bananas": 3, "mangoes": 5})
+    >>> h1+h2
+    hashdict(apples=1, bananas=3, mangoes=5)
+    >>> d1 = {}
+    >>> d1[h1] = "salad"
+    >>> d1[h1]
+    'salad'
+    >>> d1[h2]
+    Traceback (most recent call last):
+    ...
+    KeyError: hashdict(bananas=3, mangoes=5)
+
+    based on answers from
+    http://stackoverflow.com/questions/1151658/python-hashable-dicts
+
+    """
+    def __key(self):
+        return tuple(sorted(self.items()))
+
+    def __repr__(self):
+        return "{0}({1})".format(
+            self.__class__.__name__,
+            ", ".join("{0}={1}".format(str(i[0]), repr(i[1])) for i in self.__key()))
+
+    def __hash__(self):
+        return hash(self.__key())
```

Comment:
Note that this doesn't guarantee hashability - all the items (both keys and values) must be hashable for this to work... also consider using frozenset() instead of sorted(), as it is slightly cheaper in run time.